api/authenticator.py

```diff
@@ -1286,9 +1414,20 @@ def authentication_provider_document(self, library_short_name):
             }
         }
         """
+
+        login_inputs = dict(keyboard=self.identifier_keyboard)
+        if self.identifier_maximum_length:
+            login_inputs['maximum_length'] = self.identifier_maximum_length
+
+        password_inputs = dict(keyboard=self.password_keyboard)
+        if self.password_maximum_length:
+            login_inputs['maximum_length'] = self.password_maximum_length
+
         method_doc = dict(
-            labels=dict(login=unicode(self.LOGIN_LABEL),
-                        password=unicode(self.PASSWORD_LABEL))
+            labels=dict(login=unicode(_(self.identifier_label)),
+                        password=unicode(_(self.password_label))),
```

Comment:
The localization here is a nice idea, but the translation files are generated from the code, so I don't know how this would work. It might ask the translator to translate "self.password_label". We should check what bin/util/generate_translation_template does with it.

If this like doesn't cause a problem, you could put constants like `COMMON_PASSWORD_LABELS = [_("Password"), _("PIN")]` somewhere so they'll be in the file.